### PR TITLE
Docs updates for 0.10 release

### LINKED ISSUE #26: clarify that command-line import requires a channel ID, not token
Users might try and import a channel via the command line by token. Until we can address https://github.com/learningequality/kolibri/issues/3733 let's clarify the documentation that a channel ID must be passed, and describe where this ID might come from.

This was made more confusing because of this warning:

image

### LINKED ISSUE #29: in case of reports, add note about malformed databases
If users contact us with reports of malformed DBs, let's add a note somewhere that if someone sees this uncommon error, they should contact us on the forums for help recovering their data:

```
sqlite3.DatabaseError: database disk image is malformed
```

### PR DESCRIPTION
**Pull request by radinamatic**
~Still WIP, don't merge!~

End-user facing updates are ready for review!

For the new `.ini` file, and content migration commands, I'll need support in terms of deciding what goes in user docs (advanced users) and what in dev docs, and the high level list of commands and options.

cc @benjaoming @indirectlylit @lyw07 

## Various fixes
- [X]  Add warning for channel ID when importing from command line (fixes #26)
- [X] Add Material Design icons (supersedes #18)
- [X] Workaround for malformed database (fixes #29) 
- [X] Add warning for Ubuntu restricted extras (fixes [#3834](https://github.com/learningequality/kolibri/issues/3834)) 

### To document
- [x] Support for coach-specific content
- [X] Content import/export is more reliable and easier to use 
        Went through the workflow and updated screenshots and text, but since a lot of stuff was improved, I'd appreciate pointers if anything else user-facing is worth documenting.
- [x] Search has improved results and handles duplicate items
- [x] Display of answer history in learner exercises is improved
- [x] Command to migrate content directory location  [#3594](https://github.com/learningequality/kolibri/pull/3594)

### No need to document...?
- [ ] New Kolibri configuration file - **To document in 0.11 release**
- [ ] Login page is more responsive
- [ ] Windows-specific improvements and bug fixes
- [ ] Overall improved performance
- [ ] Auto-play videos

### CODE REVIEW

File: docs/manage/_get_support.rst
Comment (benjaoming) [inline docs/manage/_get_support.rst]: You can shorten `/home/<your_username>/` to `~/` both in this line and below -- the syntax of this command will not work in a Windows shell anyways. It's not a given that the `.kolibri` folder is in `/home/`, it can also be in `/var/kolibri/`

Comment (radinamatic) [reply docs/manage/_get_support.rst]: Good point @benjaoming, will change it asap! 👍

**Comment by lyw07** [comment]
@radinamatic Regarding the content migration command, we currently only have one command `kolibri manage content movedirectory <destination>` to move the content from current directory to the destination directory. 
But I'm not sure if we are going to stick with the name `movedirectory` or change to something else. Please see the discussion [here](https://github.com/learningequality/kolibri/issues/3727). Currently, the name in the [PR](https://github.com/learningequality/kolibri/pull/3594) is still `movedirectory`.
And sorry I'm not sure about the `user doc` and `dev doc` part... My guess is that we can put the usage of the command into both documentations.

**Comment by radinamatic** [comment]
Fair enough @lyw07!
Since you are still working on that PR, maybe we should start documenting it after it has been merged? 
@indirectlylit thoughts?

**Comment by indirectlylit** [comment]
> maybe we should start documenting it after it has been merged

sounds right to me. let's do it in a separate PR

File: docs/manage/_command_line.rst
Comment (indirectlylit) [inline docs/manage/_command_line.rst]: in Kolibri, we use the term 'superuser', not 'super user'

File: docs/manage/_command_line.rst
Comment (indirectlylit) [inline docs/manage/_command_line.rst]: maybe

>  you can move the Kolibri **content files** to another drive

File: docs/manage/_command_line.rst
Comment (indirectlylit) [inline docs/manage/_command_line.rst]: The commands look linux-specific, but this is a windows-looking path...

File: docs/manage/_command_line.rst
Comment (indirectlylit) [inline docs/manage/_command_line.rst:212]: This section might be misleading because it stresses "ALL Kolibri Files" and "together with the imported content channels". However, adjusting this environment variable behaves differently than the `movedirectory` command:

* Adjusting the environment variable will _not_ automatically migrate over data. You need to copy the _.kolibri_ folder manually to the new location
* If you _do_ copy the _.kolibri_ folder, the content directory will _not_ be affected if it had been previously set using the `movedirectory` command

File: docs/manage/_command_line.rst
Comment (indirectlylit) [inline docs/manage/_command_line.rst]: > in your user’s `Home` folder

This should just be 

> in your user’s home folder

without the formatting

File: docs/manage/_command_line.rst
Comment (indirectlylit) [inline docs/manage/_command_line.rst]: > in your device's `Home` directory

should be

> in your user's home directory

without the formatting

File: docs/faq.rst
Comment (indirectlylit) [inline docs/faq.rst]: I'm not sure what "(✓ ver 0.5)" means. Maybe we should delete that?

Also in the [changelog](https://github.com/learningequality/kolibri/blob/develop/CHANGELOG.rst#092), I've started using the ordering convention of: English first, everything else alphabetical

File: docs/learn.rst
Comment (indirectlylit) [inline docs/learn.rst]: > You may be different types of content available in Kolibri channels

There's a typo in this phrase

File: docs/learn.rst
Comment (indirectlylit) [inline docs/learn.rst]: This paragraph switches from singular (an exercise, it) to plural (they will be marked)

File: docs/learn.rst
Comment (indirectlylit) [inline docs/learn.rst]: note that they don't _have_ to use a hint at this point - they can just try again

File: docs/learn.rst
Comment (indirectlylit) [inline docs/learn.rst]: > you will be rewarded points, and see the suggestion to continue

I think the comma in here is unnecessary?

File: docs/manage/_command_line.rst
Comment (benjaoming) [reply docs/manage/_command_line.rst]: :+1: I think for consistency, we should keep all generic examples with POSIX paths. Another aspect is that we might not want to encourage Windows for deployments, a good way to show that is to avoid it in examples (but of course to clarify to Windows users how they need to alter commands).

    kolibri manage content movedirectory /mnt/my_external_drive/kolibri_content

File: docs/manage/_command_line.rst
Comment (benjaoming) [reply docs/manage/_command_line.rst:212]: There's an unfortunate mix of "directory" and "folder" in the terminology. I know this was never consistent in the real world, just want to ask: Do we prefer any of them, or are we fine about mixing them?

File: docs/manage/_command_line.rst
Comment (radinamatic) [reply docs/manage/_command_line.rst]: I'd love if less deployments were done on Windows, but I'm skeptical about the strategy of not documenting Windows-specific steps as means to achieve it. Knowing our target user base, I'm less worried about Linux/macOS Terminal users figuring out the way to write POSIX paths by looking at Windows-looking one, than viceversa. 

I added both examples for now.

File: docs/manage/_command_line.rst
Comment (radinamatic) [reply docs/manage/_command_line.rst:212]: Personally I prefer the term `folder`, but I don't see a clear path to get out of the terminology "mix" for now... 😕 

Added Devon's suggestions in a note.

File: docs/faq.rst
Comment (radinamatic) [reply docs/faq.rst]: Kolibri was localized into Mexican Spanish for version 0.5 last year (UNETE launch), but that was never updated. As per Lauren's suggestion, I left only the check marks, but I missed the suggestion about ordering convention - will amend!

File: docs/manage/_command_line.rst
Comment (radinamatic) [reply docs/manage/_command_line.rst]: I think I made that change sometimes in the last version update, but now I can't seem to find the reputable source that prompted that decision. I don't have a strong preference for either option, and it's hard to gauge the preference by Google search. So **Superuser** or **Super user**? 🙂 

cc @khangmach @jtamiace @laurenlichtman 

On the side note: this would be a good question to jumpstart the `UI writing` discussion space we previously mentioned...

Comment (khangmach) [reply docs/manage/_command_line.rst]: I've been using the term "superuser" in all documentation and it seems the term also already appears in Kolibri as "superuser". I think the special connotation behind "superuser" might also get lost in various translations. The way I see it is: the user isn't just a user that is "super", they have a special label called "superuser"

Comment (laurenlichtman) [reply docs/manage/_command_line.rst]: I agree with @khangmach - it's not a user that is super, it's a noun and a specific role in Kolibri. I vote "superuser."

File: docs/contact.rst
Comment (indirectlylit) [inline docs/contact.rst]: just a heads-up that the title is a bit long and will wrap. not sure if this is a concern or not, but aesthetically not optimal

image

image

Comment (radinamatic) [reply docs/contact.rst]: Thoughts @carinediaz @laurenlichtman ? ^^

Comment (benjaoming) [reply docs/contact.rst]: Why can't it be called "Getting Support"?

Comment (radinamatic) [reply docs/contact.rst]: My guess is that @carinediaz wanted to put more emphasis on community support...

Comment (benjaoming) [reply docs/contact.rst]: I think it's okay to leave it at "Getting support" since that would be the objective of the one reading the navigation, then immediately explaining about community, which is also somewhat okay now -- but can of course be optimized in many ways :)

For instance, we could add a "Support others" sort of sub-heading or just a bold text in a paragraph below:

> If you are looking for support, please consider to **support others**. Your goals are likely overlapping with other users of Kolibri, so you may consider yourself as part of a community of like-minded practitioners!

Comment (carinediaz) [reply docs/contact.rst]: @radinamatic @benjaoming I just changed to Get in Touch bc if I remember correctly, we were repeating Getting Support twice on the page. You can edit it back to `Getting Support`, no worries, let's just change the subheading to 'How to get support from Learning Equality and from our community'?

Comment (benjaoming) [reply docs/contact.rst]: Any thoughts on the text that I drafted above?

> If you are looking for support, please consider to support others. Your goals are likely overlapping with other users of Kolibri, so you may consider yourself as part of a community of like-minded practitioners!

Comment (radinamatic) [reply docs/contact.rst]: This is my current draft, Ben's text is mostly in, subheading in one line:

File: docs/coach.rst
Comment (indirectlylit) [inline docs/coach.rst]: I don't think the new version is syntactically correct. Was the intent "Coaches' Support Resources"?

Without more context on why it was changed, the most clear way I can think of writing this header would be "Supporting Resources for Coaches".

File: docs/install/_system_requirements.rst
Comment (indirectlylit) [inline docs/install/_system_requirements.rst]: > In case you are running Kolibri on Linux and want an efficient setup

This sounds correct

> it's required to use if you implement on a system with specs equivalent to or smaller than Raspberry Pi.

'Implement' sounds a bit awkward here - I don't think anyone would actually say this sentence out loud.

Proposal:

* When describing scenarios where humans are affected by our products, 'implement' will often work
* For describing purely technical processes, use words like 'install' or 'run'

Comment (benjaoming) [reply docs/install/_system_requirements.rst]: The package `kolibri-raspberry-pi` does not exist - I commented that in the Google Doc, flagging here also: Can we remove all of the note?

Comment (benjaoming) [reply docs/install/_system_requirements.rst]: ...it doesn't exist yet. But it's confusing to mention it now, people will spend time looking for it.

Comment (radinamatic) [reply docs/install/_system_requirements.rst]: Noted, will comment out the note until we are ready to offer the package,

File: docs/coach.rst
Comment (radinamatic) [reply docs/coach.rst]: My mistake, bad copy/paste from [Navya's suggestion in the review doc](https://docs.google.com/document/d/1mx3SRLb2huPggC_D3blzeIkAcxch3iOUm_VBHozxVoQ/edit#bookmark=id.qk5n8gg07ljg).

File: docs/install/_linux.rst
Comment (indirectlylit) [inline docs/install/_linux.rst]: we may want to say `v0.10.0` or `v0.10.x` rather than `v0.10` to avoid confusion

Comment (indirectlylit) [reply docs/install/_linux.rst]: or, maybe not?

Comment (radinamatic) [reply docs/install/_linux.rst]: I think the latest recommendation from @imps was `v0.10`. 
cc @laurenlichtman

Comment (indirectlylit) [reply docs/install/_linux.rst]: fine with me, as long as it doesn't confuse people to occasionally see it in both forms

Comment (radinamatic) [reply docs/install/_linux.rst]: I see what you mean... I find `v0.10.x` a bit verbose inline (as opposed to commands that require precise filenames), but will defer to what @imps believe is best.

File: docs/_static/theme_overrides.css
Comment (benjaoming) [inline docs/_static/theme_overrides.css:1]: This makes our docs online-only. They are intended to be distribute-able offline. What is the licens of this icon stuff, are we permitted to copy it into our own source?

File: docs/_static/theme_overrides.css
Comment (benjaoming) [inline docs/_static/theme_overrides.css:36]: This affects all images, but is it what we want? It will make a smaller image stretched beyond its resolution.

There's already `max-width: 100%`, so I would suggest deleting this.

Comment (benjaoming) [reply docs/_static/theme_overrides.css:36]: What was the purpose of this?

File: docs/access.rst
Comment (benjaoming) [inline docs/access.rst]: I'm not sure that works and I have no Mac to test it. However, `python -m kolibri start` will work.

File: docs/access.rst
Comment (benjaoming) [inline docs/access.rst:71]: Remember the obscure rule of using the double underscore in links :)

**Review by benjaoming — CHANGES_REQUESTED** [review]
I'm going to make a direct PR with my changes :)

File: docs/access.rst
Comment (benjaoming) [inline docs/access.rst]: Is this really true? Is there an issue open about it? Selecting language from HTTP request headers sent by the browser isn't the only way to do things :)

(I'm sending some changes to this paragraph btw)

File: docs/_static/theme_overrides.css
Comment (radinamatic) [reply docs/_static/theme_overrides.css:1]: Yes, for now it's online only, same as our docs, but a set of icons can easily be included once we start exploring the option of making offline docs.

License is not the issue, as these are exactly the same icons we use in Kolibri.

File: docs/access.rst
Comment (radinamatic) [reply docs/access.rst]: Note was requested by the @imps team, as it happened to one of them when trying to start the PEX on her computer.

cc @indirectlylit

File: docs/access.rst
Comment (radinamatic) [reply docs/access.rst]: There is a clearinghouse issue (#219), and seen in practice while performing the upgrade with VF partners (cc @jamalex).

Comment (benjaoming) [reply docs/access.rst]: Thanks, posted a comment there.

File: docs/access.rst
Comment (indirectlylit) [reply docs/access.rst]: > Note was requested by the @imps team, as it happened to one of them when trying to start the PEX on her computer.

Above, you have a separate bullet for "If you used the PEX package..."

In the case of PEX, users need to make sure they:

```bash
chmod +x kolibri-v0.10.0.pex
./kolibri-v0.10.0.pex start  # note './'
```

The advice given might work, but it's confusing to me that in the case of a PEX there would be something called `kolibri` accessible on the path.

Comment (indirectlylit) [reply docs/access.rst]: (also note this advice appears to be under the section for PPA and Pip)

Comment (benjaoming) [reply docs/access.rst]: @indirectlylit it's valid if someone installs on Mac via `pip install kolibri`, that's the only case I can think of. But I still think it would have to be with the `-m` switch to call the `kolibri` _module_ in cases where the command doesn't enter the PATH environment.

File: docs/_static/theme_overrides.css
Comment (benjaoming) [reply docs/_static/theme_overrides.css:1]: Alright, cool, then let's open an issue to put this directly in the docs as static assets :)

Comment (radinamatic) [reply docs/_static/theme_overrides.css:1]: Yes, not necessary to include in this release (we might need to make a subset of MD icons, no need to include all), but as soon as offline docs are a reality, the issue to remind us will be handy.

File: docs/_static/theme_overrides.css
Comment (radinamatic) [reply docs/_static/theme_overrides.css:36]: It affects only inline images and was introduced for better spacing.

Without:


With:

**Pull request merged**
